components/brave_wallet_ui/utils/tx-utils.ts

```diff
@@ -1962,3 +1965,17 @@ export function getTransactionTypeName(txType: BraveWallet.TransactionType) {
       return getLocale('braveWalletTransactionTypeNameOther')
   }
 }
+
+export const getIsSolanaAssociatedTokenAccountCreation = (
+  transaction: TransactionInfo,
+) => {
+  const txCoinType = getCoinFromTxDataUnion(transaction.txDataUnion)
+  return (
+    txCoinType === BraveWallet.CoinType.SOL
+    && transaction.txType
+      === BraveWallet.TransactionType.SolanaDappSignAndSendTransaction
+    && transaction.txDataUnion.solanaTxData?.staticAccountKeys.includes(
+      BraveWallet.SOLANA_ASSOCIATED_TOKEN_PROGRAM_ID,
```

Comment:
Wondering if it is a good idea to use an `includes` here. There are a lot of transactions that may have ATA creation step as one of its instructions. We should not indicate that the transaction is solely creating an ATA for such cases.